test/docker-e2e/e2e_upgrade_test.go

```diff
@@ -0,0 +1,294 @@
+package docker_e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"celestiaorg/celestia-app/test/docker-e2e/dockerchain"
+
+	signaltypes "github.com/celestiaorg/celestia-app/v5/x/signal/types"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	tastoradockertypes "github.com/celestiaorg/tastora/framework/docker"
+	"github.com/celestiaorg/tastora/framework/testutil/wait"
+	tastoratypes "github.com/celestiaorg/tastora/framework/types"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+)
+
+// TestCelestiaAppBinarySwapUpgrade tests a simple upgrade from one image tag
+// to another by swapping the binary (usually used for minor version upgrade)
+func (s *CelestiaTestSuite) TestCelestiaAppBinarySwapUpgrade() {
+	if testing.Short() {
+		s.T().Skip("skipping celestia-app minor upgrade test in short mode")
+	}
+
+	tt := []struct {
+		Name           string
+		BaseImageTag   string
+		TargetImageTag string
+	}{
+		{
+			Name:           "v4.0.2-rc2 to v4.0.10",
+			BaseImageTag:   "v4.0.2-rc2",
+			TargetImageTag: "v4.0.10",
+		},
+		{
+			Name:           "v4.0.9-mocha to v4.0.10-mocha",
+			BaseImageTag:   "v4.0.9-mocha",
+			TargetImageTag: "v4.0.10-mocha",
+		},
+		{
+			Name:           "v4.0.9-arabica to v4.0.10-arabica",
+			BaseImageTag:   "v4.0.9-arabica",
+			TargetImageTag: "v4.0.10-arabica",
+		},
+	}
```

Comment:
blocking

sorry, I'm still missing something. This looks like its ran with the other tests, is that true?

 This looks like its using the same image each run, is that true? 

if false, are we downgrading from the existing branch to these?

if true, what benefit is there to running it more than once?

Reply:
> If we decide they are too heavy for the default pipeline we could move them to the nightly workflow or plan it the way that is needed.

its not strictly that they're too heavy, but just that if we're not incorporating the existing branch from the PR that is being tested then there's no point to run the test. It's completely unrelated to the PR.

> Right now I am removing this minor restart test. So only signal upgrade will be running.

thank you!!